packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -39,6 +39,7 @@ class InputDateTimePicker extends React.Component {
 			PropTypes.number,
 			PropTypes.string,
 		]),
+		onFinish: PropTypes.func,
```

Comment:
If you introduce the button and the automatic finish if no time, you should change the way changes are passed to the outside world. You don't need a `props.onFinish()`

This means that 
* `this.onPickerChange` updates only the internal state + call `this.onFinish()` if no time
* `this.onFinish()` call `props.onChange()` and close the picker

This has an impact on form widget
* onChange : trigger form onFinish() if it comes from picker
* on input blur : trigger form onFinish()

There is another behavior that this button implies. It seems to be a validation button, so it means that if we don't click on it, but close the picker (esc for example), it should not apply the `temporary value` ? Let's check with UX on this one too.

Reply:
for the last sentence i guess yes.